### PR TITLE
[V1] Fix torch profiling for offline inference

### PR DESCRIPTION
**Pull request by ywang96**
Tested with `VLLM_USE_V1=1 python3 examples/offline_inference_with_profiler.py`. Without this fix, running the same script will error out with
```
RuntimeWarning: coroutine 'InprocClient.profile' was never awaited
```

**Edit:**
Also fixed for `VLLM_USE_V1=1 VLLM_ENABLE_V1_MULTIPROCESSING=1 python3 examples/offline_inference_with_profiler.py`. Two notable changes are:
- We need to wrap the existing code with `if __name__ == "__main__":` to make it work with MP/`SyncMPClient`.
- A sleep time buffer was added so that the main process won't terminate too early for the background process to finish writing the profiling output. (This is somewhat hacky but I don't know if there's a better way to do so)

### CODE REVIEW

**Comment by ywang96** [comment]
Hello @Abatom! It looks like you add the support for profiling with V1 in #10564, did you test it with all three clients?

**Review by DarkLight1337 — APPROVED** [review]

**Comment by Abatom** [comment]
> Hello @Abatom! It looks like you added the support for profiling with V1 in #10564, did you test it with all three clients? (`InprocClient`, `SyncMPClient`, `AsyncMPClient`)

Sorry, I didn't test everything.

File: examples/offline_inference_with_profiler.py [hunk 1]
```diff
@@ -15,19 +16,25 @@
 # Create a sampling params object.
 sampling_params = SamplingParams(temperature=0.8, top_p=0.95)
 
-# Create an LLM.
-llm = LLM(model="facebook/opt-125m", tensor_parallel_size=1)
+if __name__ == "__main__":
```
Comment (tlrmchlsmth) [inline examples/offline_inference_with_profiler.py:19]: Note that you shouldn't need the `if __name__ == "__main__":` once https://github.com/vllm-project/vllm/pull/11074 lands

**Review by tlrmchlsmth — APPROVED** [review]
Thanks for fixing this!

**Pull request merged**